do.go

```diff
@@ -16,13 +16,45 @@ func setPath() (unset func()) {
 	}
 }
 
+func setGoEnv() (unset func()) {
+	prevGoPath, goPathWasSet := os.LookupEnv("GOPATH")
+	var newGoPath string
+	if goPathWasSet {
+		newGoPath = prevGoPath + string(os.PathListSeparator) + toolDirPath
+	} else {
+		newGoPath = toolDirPath
+	}
+	_ = os.Setenv("GOPATH", newGoPath)
+
+	prevGoBin, goBinWasSet := os.LookupEnv("GOBIN")
+	newGoBin := filepath.Join(toolDirPath, "bin")
+	_ = os.Setenv("GOBIN", newGoBin)
+
+	return func() {
+		if goPathWasSet {
+			_ = os.Setenv("GOPATH", prevGoPath)
+		} else {
+			_ = os.Unsetenv("GOPATH")
+		}
+		if goBinWasSet {
+			_ = os.Setenv("GOBIN", prevGoBin)
+		} else {
+			_ = os.Unsetenv("GOBIN")
+		}
+	}
+}
+
 func do() {
 	args := positionalArgs
 	if len(args) == 0 {
 		fatal("no command passed to retool do", nil)
 	}
 
-	defer setPath()()
+	unsetPath := setPath()
+	defer unsetPath()
+
+	unsetGoEnv := setGoEnv()
```

Comment:
Why are you un-setting the environment? Surely you are not modifying the parent's environment so when retool is done these changes will be gone? I might be missing something.